### ISSUE TITLE
Numbering pre-check: grep -oP fails to an empty result on Git-Bash, proposing Observation 1 in a populated log

### ISSUE THREAD

**Issue by rlucho** [rebelytics/one-skill-to-rule-them-all#25]
## Summary

The mandatory numbering pre-check in **Observation Protocol → How to Log** leads with a PCRE `grep` form that fails on some hosts **to an empty stdout rather than to an error**. When it fails, the skill proposes **Observation 1** in a log whose highest entry is in the dozens.

## The failing command

```bash
# GNU grep (Linux, Cowork):
grep -oP '### Observation \K\d+' log.md | sort -n | tail -1

# macOS / POSIX-compatible alternative:
grep -o '### Observation [0-9]*' log.md | grep -o '[0-9]*' | sort -n | tail -1
```

On Windows + Git Bash (and any host whose locale isn't unibyte/UTF-8), the first form fails with:

```
grep: -P supports only unibyte and UTF-8 locales
```

That message goes to **stderr**; **stdout is empty**. So `HIGH` is empty, `$((HIGH + 1))` evaluates to **1**, and the skill proposes `### Observation 1:`.

## Why the existing safety nets don't fully cover it

- The **write-time collision assertion** does catch it at append time — which is exactly what it's for, and it worked. But it only fires at the moment of writing.
- The same silent-empty failure also occurs during the **session-start scan** (Session Start Protocol step 2), where it simply reports "no observations found" and **nothing flags it**. The agent starts the session believing the log is empty.

The skill's own wording — "never rely on session memory, always read the actual log file" — is satisfied by a command that reads the file and silently returns nothing.

## Suggested fix

**1. Promote the POSIX form to the primary command.** It works on GNU, BSD/macOS and Git-Bash alike, so the platform-specific note is unnecessary:

```bash
HIGH=$(grep -o '### Observation [0-9]*' log.md | grep -o '[0-9]*' | sort -n | tail -1)
```

**2. Require the pre-check to fail loudly on an empty result**, so "the file says zero" is distinguishable from "I could not read the file":

```bash
if [ -z "$HIGH" ] && [ -s log.md ]; then
  echo "NUMBERING COMMAND BROKEN — log.md is non-empty but no numbers extracted"; exit 1
fi
```

**3. Same substitution in the other two occurrences** — the write-time assertion (`PROPOSED=…`) and the post-write renumber block (`NEW_NUM=…`) both use `grep -oP` too.

**4. Related, same class of bug:** the post-write renumber block uses `sed -i "…" log.md`. GNU `sed` requires no argument after `-i`; BSD/macOS `sed` requires one. A portable in-place edit avoids `-i` entirely:

```bash
sed "${MY_LINE}s/^### Observation ${PROPOSED}:/### Observation ${NEW_NUM}:/" \
  log.md > log.md.tmp && mv log.md.tmp log.md
```

## Generalisable principle

A skill that hands the agent a shell command owns that command's portability. Where a fallback exists, **lead with the portable form** — an alternative offered as a platform-specific footnote is one the agent reaches for only *after* the primary has already failed. And a command that fails to *empty* rather than to *error* may never announce that it failed at all, so any computation that derives a number from a file should distinguish "the file says zero" from "I could not read the file".

## Environment

- Windows 11, Git Bash, `grep` (GNU grep) with a non-UTF-8 locale
- Skill used via Claude Code with the recommended CLAUDE.md activation instruction

Happy to open a PR with the four edits if that's useful. Thanks for the skill — the observation/review loop has been genuinely valuable.

**Comment by rebelytics** [comment]
Thanks for this, @rlucho. I will fix it in v3 and make you a co-author of the PR, so you’re credited for the improvement.

**Comment by rebelytics** [comment]
Fixed in `release/v3.0.0`, partly by the storage change: the `grep -oP` numbering blocks are gone, because 3.0.0 stores one file per observation and the id comes from the directory listing. What survives from your report is the rule: the id computation fails loudly when the log is populated but no ids were extracted, so "the file says zero" and "I could not read the file" are no longer the same result.

You're credited as co-author on the commit. FYI: I'll be running v3 for a week or so before it goes into main.

**Comment by rebelytics** [comment]
Shipped in v3.0.0, released today. Thanks again for your contribution — the release notes credit everyone whose reports and PRs shaped this version.
